Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -347,21 +347,46 @@ WASM_SLOW_PATH_DECL(array_new_default)
     WASM_RETURN(Wasm::operationWasmArrayNew(instance, instruction.m_typeIndex, size, value));
 }
 
-WASM_SLOW_PATH_DECL(array_get)
+template <typename WasmOp>
+static auto ArrayGetOperation(CallFrame* callFrame, const WasmInstruction* pc, Wasm::Instance* instance, std::function<EncodedJSValue(Wasm::StorageType, EncodedJSValue)> callback)
 {
-    auto instruction = pc->as<WasmArrayGet>();
+    auto instruction = pc->as<WasmOp>();
     EncodedJSValue arrayref = READ(instruction.m_arrayref).encodedJSValue();
     if (JSValue::decode(arrayref).isNull())
         WASM_THROW(Wasm::ExceptionType::NullArrayGet);
     uint32_t index = READ(instruction.m_index).unboxedUInt32();
-
     JSValue arrayValue = JSValue::decode(arrayref);
     ASSERT(arrayValue.isObject());
     JSWebAssemblyArray* arrayObject = jsCast<JSWebAssemblyArray*>(arrayValue.getObject());
     if (index >= arrayObject->size())
         WASM_THROW(Wasm::ExceptionType::OutOfBoundsArrayGet);
+    WASM_RETURN(callback(arrayObject->elementType().type, Wasm::operationWasmArrayGet(instance, instruction.m_typeIndex, arrayref, index)));
+}
+
+WASM_SLOW_PATH_DECL(array_get)
+{
+    return ArrayGetOperation<WasmArrayGet>(callFrame, pc, instance, [=](auto, auto result) {
+        return result;
+    });
+}
+
+WASM_SLOW_PATH_DECL(array_get_s)
+{
+    auto callback = [=](Wasm::StorageType type, EncodedJSValue value) {
```

Comment:
The signed array get processing is now done via a callback here (no-op callback in the other cases).